### ISSUE TITLE
Assigned members to managed event types missing preview URL

### ISSUE THREAD

**Issue by leog** [calcom/cal.com#8386]
### Issue Summary

When creating a managed event type, assigning team members in the Assignment section, once the changes are saved, the "Preview URL" button for each member does not appear. Only when refreshing the page.

This is likely due to not able to sync backend with frontend in the scheme of components needed to show the list of team members assigned.

### Steps to Reproduce

1. Create manage event type
2. Go to Assignment tab within the newly managed event type created
3. Assign a team member
4. Hit "Save" button

Notice the preview URL button is not shown for each assign member unless you refresh the page.

You can log in with teampro@example.com username and teampro password and assign already existent members like teampro2.

**Comment by Toledodev** [comment]
Hey @leog, the preview button should appear only after the user click on save button? I made some changes and now the preview button appears when we select the team member
Please take a look: https://www.loom.com/share/16571e378aca469d8cc4912513fba648

**Comment by leog** [comment]
@Toledodev right, only when the team member assigned is actually assigned after saving, before that the preview link would not work
